### PR TITLE
`!!`: Fix extra `\r` on Windows

### PR DESCRIPTION
**Pull request by kazarmy**
**Your checklist for this pull request**
- [X] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [X] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [X] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

This pr prevents the `\r\r\n` sequence from occurring in `!!` output on Windows, by converting `\r\n` to `\n` in the output of the client process.

**Test plan**

All builds are green.

**Closing issues**

...

### CODE REVIEW

File: librz/core/cmd/cmd_system.c [hunk 1]
```diff
@@ -191,6 +191,21 @@ static RzCmdStatus system_common_handler(RzCore *core, bool force_rzcons, int ar
 	bool succ = system_exec(core, argc - 1, &argv[1], need_rzcons ? &out : NULL, &length, &ret);
 	rz_cons_sleep_end(bed);
 	if (need_rzcons) {
+#if __WINDOWS__
+		char *src = out;
+		char *dest = src;
+		char *end = out + length;
+		while (src != end) {
+			*dest = *src;
+			if (src[0] == '\r' && src[1] == '\n') {
```
Comment (ret2libc) [inline librz/core/cmd/cmd_system.c]: is `src[1]` always within the bounds?

**Review by ret2libc — COMMENTED** [review]
Why is the \r\r\n happening? what's the problem with that?

File: librz/core/cmd/cmd_system.c (hunk 1 above)
Comment (kazarmy) [reply librz/core/cmd/cmd_system.c]: hmm good catch ceaca2e217cdb516d10ac52e6aa2ca5d75c766d7

**Comment by kazarmy** [comment]
> Why is the \r\r\n happening? what's the problem with that?

On Windows, when `!!` captures the output of a process, the output lines can be terminated with `\r\n` as a result of normal stdio printing ... when rz_cons then tries to print the captured output using `write()` (alias of [`_write()`](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/write?view=msvc-160)), it helpfully converts `\n` into `\r\n`, thus `\r\n` becomes `\r\r\n`[^1]. 




The problem here is that the additional `\r` isn't added by the process but by Windows, and scripts shouldn't really need to treat Windows `!!` output differently from POSIX `!!` output.

[^1]: The relevant line in the `_write()` documentation is:
  "If the file is opened in text mode, each line feed character is replaced with a carriage return-line feed pair in the output."
  It says nothing about converting carriage return characters.

**Comment by kazarmy** [comment]
> ... as a result of normal stdio printing

This is because stdout [defaults to text mode](https://docs.microsoft.com/en-us/cpp/c-runtime-library/text-and-binary-mode-file-i-o?view=msvc-160). What text mode does is stated [here](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/setmode?view=msvc-160#remarks).

**Review by ret2libc — APPROVED** [review]

**Pull request merged**